crates/illumos-sys-hdrs/src/lib.rs

```diff
@@ -127,37 +131,70 @@ impl Default for kstat_named_t {
         Self {
             name: [0; KSTAT_STRLEN],
             dtype: 0,
-            value: KStatNamedValue { _c: [0; 16] },
+            value: KStatNamedValue(UnsafeCell::new(_KStatNamedValue {
+                _c: [0; 16],
+            })),
         }
     }
 }
 
 #[repr(C)]
-pub union KStatNamedValue {
+pub union _KStatNamedValue {
     _c: [c_char; 16],
     _i32: i32,
     _u32: u32,
     _i64: i64,
     _u64: u64,
 }
 
+#[repr(transparent)]
+pub struct KStatNamedValue(UnsafeCell<_KStatNamedValue>);
+
 impl core::ops::AddAssign<u64> for KStatNamedValue {
     #[inline]
     fn add_assign(&mut self, other: u64) {
-        unsafe { self._u64 += other };
+        self.incr_u64(other);
     }
 }
 
 impl core::ops::SubAssign<u64> for KStatNamedValue {
     #[inline]
     fn sub_assign(&mut self, other: u64) {
-        unsafe { self._u64 -= other };
+        self.decr_u64(other);
     }
 }
 
 impl KStatNamedValue {
-    pub fn set_u64(&mut self, val: u64) {
-        self._u64 = val;
+    /// Validates at compile time whether `self._u64` can be safely used as
+    /// an AtomicU64.
+    const KSTAT_ATOMIC_U64_SAFE: () = if align_of::<KStatNamedValue>() % 8 == 0
+    {
+    } else {
+        panic!("Platform does not meet u64 8B alignment for AtomicU64");
+    };
+
+    #[inline(always)]
+    #[allow(clippy::let_unit_value)]
+    fn as_u64(&self) -> &AtomicU64 {
+        _ = Self::KSTAT_ATOMIC_U64_SAFE;
+        // SAFETY: KStatNamedValue must have 8B alignment on target platform.
+        //         Validated by compile time check in `KSTAT_ATOMIC_U64_SAFE`.}
```

Comment:
Nit, stray `}` appeared:
```suggestion
        //         Validated by compile time check in `KSTAT_ATOMIC_U64_SAFE`.
```

Reply:
Done in 587082d.